lib/insights/api/common/application_controller_mixins/exception_handling.rb

```diff
@@ -7,19 +7,25 @@ module ExceptionHandling
 
           def self.included(other)
             other.rescue_from(StandardError, RuntimeError) do |exception|
-              logger.error("#{exception.class.name}: #{exception.message}\n#{exception.backtrace.join("\n")}")
-              errors = Insights::API::Common::ErrorDocument.new.tap do |error_document|
-                exception_list_from(exception).each do |exc|
-                  if api_client_exception?(exc)
-                    api_client_errors(exc, error_document)
-                  else
-                    error_document.add(error_code_from_class(exc).to_s, "#{exc.class}: #{exc.message}")
-                  end
-                end
+              rescue_from_handler(exception) do |error_document, exc|
+                error_document.add(error_code_from_class(exc).to_s, "#{exc.class}: #{exc.message}")
               end
+            end
+          end
 
-              render :json => errors.to_h, :status => error_code_from_class(exception)
+          def rescue_from_handler(exception)
+            logger.error("#{exception.class.name}: #{exception.message}\n#{exception.backtrace.join("\n")}")
+            errors = Insights::API::Common::ErrorDocument.new.tap do |error_document|
+              exception_list_from(exception).each do |exc|
+                if api_client_exception?(exc)
```

Comment:
@eclarizio can we check if the class name is PunditAuthorizationException and tweak the message here like we do with api exceptions from generated clients. This way it would help all other users of Pundit.

Reply:
I suppose we could, but doesn't that then make this gem sort of depend upon pundit just for the exception declaration? I don't know if I like that idea.

Reply:
I'm ok with checking the class name there, we do it for all of the generated clients already so another check fits in line. Both Catalog and Approval are using pundit so far right?